### PR TITLE
Add a simple baseline test for the sample app

### PR DESCRIPTION
**Pull request by badboy**
Same test as the android sample app

* [x] Wait for #442 to land
* [x] Enable on CI

### CODE REVIEW

File: samples/ios/app/glean-sample-app/Info.plist [hunk 1]
```diff
@@ -41,5 +41,12 @@
 		<string>UIInterfaceOrientationLandscapeLeft</string>
 		<string>UIInterfaceOrientationLandscapeRight</string>
 	</array>
+	<key>NSAppTransportSecurity</key>
```
Comment (Dexterp37) [inline samples/ios/app/glean-sample-app/Info.plist]: nit: can we add a comment here mentioning why these permissions are required? (i.e. sending pings to a local server on UI tests)

File: samples/ios/app/glean-sample-app/AppDelegate.swift [hunk 2]
```diff
@@ -13,7 +13,16 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
     func application(_: UIApplication, didFinishLaunchingWithOptions _: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
         let glean = Glean.shared
         glean.setUploadEnabled(true)
-        glean.initialize()
+
+        if ProcessInfo.processInfo.arguments.contains("USE_MOCK_SERVER") {
+            let address = "http://localhost:9080"
+            print("using a mock server, setting address: \(address)")
```
Comment (Dexterp37) [inline samples/ios/app/glean-sample-app/AppDelegate.swift]: Is this and the one below meant to be a `printf`? Does Apple use any other logging mechanism that we should use here?

File: samples/ios/app/glean-sample-appUITests/BaselinePingTest.swift [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        // In UI tests it is usually best to stop immediately when a failure occurs.
+        continueAfterFailure = false
+
+        // UI tests must launch the application that they test.
+        // Doing this in setup will make sure it happens for each test method.
+        app = XCUIApplication()
+    }
+
+    override func tearDown() {
+        server?.stop()
```
Comment (Dexterp37) [inline samples/ios/app/glean-sample-appUITests/BaselinePingTest.swift]: Is this being called after each test? If so, should we call `setupServer` in `setUp`? (or comment on why this is not being done)

File: bin/run-ios-sample-app-build.sh [hunk 4]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
```
[the remark this answers is not in the document]
Comment (Dexterp37) [reply bin/run-ios-sample-app-build.sh]: (it's fine to do this as a follow-up or file a bug to add it to all bash scripts)

[the remark this answers is not in the document]
Comment (badboy) [reply bin/run-ios-sample-app-build.sh]: *assigns it to the public domain* ... but actually I'm filing a bug to figure this out

File: samples/ios/app/glean-sample-app/Info.plist (hunk 1 above)
Comment (badboy) [reply samples/ios/app/glean-sample-app/Info.plist]: This file is edited through the GUI. I'm not sure how nice it plays with adding comments :/

File: samples/ios/app/glean-sample-app/AppDelegate.swift (hunk 2 above)
Comment (badboy) [reply samples/ios/app/glean-sample-app/AppDelegate.swift]: `print` it is! (or `NSLog`, which is slower).  We use the same in the Logger class in Glean.

As this is just the sample app I didn't want to add more sophisticated logging.

File: samples/ios/app/glean-sample-appUITests/BaselinePingTest.swift (hunk 3 above)
Comment (badboy) [reply samples/ios/app/glean-sample-appUITests/BaselinePingTest.swift]: This is by now mostly a left-over of my previous code.
I'm refactoring the server function in #444, would you okay with letting this slip here and I move around the `server.stop` over in the follow up?

Comment (Dexterp37) [reply samples/ios/app/glean-sample-appUITests/BaselinePingTest.swift]: Yup! That's fine, thanks!

File: samples/ios/app/glean-sample-app/Info.plist (hunk 1 above)
Comment (Dexterp37) [reply samples/ios/app/glean-sample-app/Info.plist]: Oh, eugh. Ok! No problem then!

Comment (badboy) [reply samples/ios/app/glean-sample-app/Info.plist]: Just tested it. Xcode will removed the comment if someone opens and saves the Info.plist inside Xcode.

**Pull request merged**